### PR TITLE
Improved yarn rwt link, with selective copy

### PR DESCRIPTION
**Pull request by dac09**
### What?
As discussed, new workflow:
 - copies only files that are meant to be packaged from `FRAMEWORK_PATH/packages/` to `PROJECT_PATH/redwood`
 - attempted to use esbuild, but unfortunately can't do that just yet. However code is simple enough to switch over when this is possible
 - link is now a simple nodejs script, does not need to be compiled before to use, so we can run it from the repo too like this:
```
yarn build:link -- -- --dest PATH_TO_PROJECT/packages
```
(crazy dashes for lerna)
- if you're a super advanced user you can do selective link like this
```
cd packages/cli
./../../tasks/build-link --dest PATH_TO_PROJECT/packages
```
This will only link the cli package. But I haven't exposed this in rwt cli.

### CODE REVIEW

**Comment by dac09** [comment]
The error may not be related to anything we're doing, setting policies like this seems to help: https://github.com/yarnpkg/yarn/issues/8405

File: packages/cli/src/redwood-tools.js [hunk 1]
```diff
@@ -81,7 +81,7 @@ export const fixProjectBinaries = (PROJECT_PATH) => {
     })
 }
 
-export const copyFiles = async (src, dest) => {
+export const copyFiles = async (src, dest, { skipChmod, silent } = {}) => {
   // TODO: Figure out if we need to only run based on certain events.
```
Comment (dac09) [inline packages/cli/src/redwood-tools.js:85]: Ignore these changes, I'll take them out

File: packages/cli/src/redwood-tools.js [hunk 2]
```diff
@@ -153,41 +155,47 @@ const rwtLink = async (yargs) => {
...
-  fs.symlinkSync(frameworkPackagesPath, symLinkPath)
-  updateProjectWithResolutions(frameworkPath)
+  fs.mkdirSync(projectPackagesPath)
+
+  updateProjectWithResolutions(frameworkPackagesPath)
+
+  addRedwoodFolderToGitIgnore()
+
+  // Let workspaces do the link
+  await execa('yarn build:link', [' -- -- --dest', projectPackagesPath], {
```
Comment (dac09) [inline packages/cli/src/redwood-tools.js:172]: This is the key change

File: tasks/build-link [hunk 3]
```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env node
```
Comment (dac09) [inline tasks/build-link:1]: And this is how we're copying only relevant files, and using esbuild to build the framework. There must be some config missing in esbuild

**Comment by dac09** [comment]
@peterp ready for review. Unfortunately couldn't get esbuild to work, as we discussed, but hopefully in the future!

**Review by thedavidprice — APPROVED** [review]
I trust whatever @dac09 wants to do ¯\_(ツ)_/¯

**Comment by dac09** [comment]
Merging to test with canary, happy to revert if there's objections later

**Comment by thedavidprice** [comment]
@dac09 Forgot to ask --> does this change the "how to use" instructions/flow?

**Comment by dac09** [comment]
Nope, non-breaking change :) Just make sure your local branch has the latest code from main merged :)

**Comment by thedavidprice** [comment]
Trying it now!

**Comment by thedavidprice** [comment]
Not working for me. Here's my process:
1. installed canary in my local Dev Project (DP)
2. pulled latest code into local Framework (F)
3. F: clean and install
4. DP: `yarn rwt link path/to/F`
5. waited 'till things stopped moving
6. DP: `yarn rw help` ERROR "no such file or directory"

😢

**Comment by dac09** [comment]
> Not working for me. Here's my process:
> 
> 1. installed canary in my local Dev Project (DP)
> 
> 2. pulled latest code into local Framework (F)
> 
> 3. F: clean and install
> 
> 4. DP: `yarn rwt link path/to/F`
> 
> 5. waited 'till things stopped moving
> 
> 6. DP: `yarn rw help` ERROR "no such file or directory"
> 
> 
> 
> 😢 

There's a chance you will need to wait a little longer! I've noticed that the console output "stops" for a bit while it's doing something, then resumes. 

Best way is to wait till you see the box that tells you "your project is now linked!"

Hope this fixes it for you!!

**Comment by thedavidprice** [comment]
> Best way is to wait till you see the box that tells you "your project is now linked!" (with 🚀 emoji ofcourse)

^^ ha! This never displayed for me. I did give it quite awhile. Can try again.

**Pull request merged**